### DISCUSSION TITLE
Getting zone/point/outline coordinates

### DISCUSSION THREAD

**Discussion by PiotrMachowski** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#318] [FAQ]
<h2>Since v2.1.0 there is a simplified way to get coordinates for selection:</h2>

[Documentation](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/blob/master/docs/templates/setup.md)

<details>

<summary><h2>Getting selection before v2.1.0</h2></summary>

To get coordinates of zone, point or outline you have to:
1. Select appropriate mode:
    - zone => Zones (`template: vacuum_clean_zone`)
    - point => Manual Point (`template: vacuum_goto`)
    - outline => Path (`template: vacuum_follow_path`) 

    If your platform doesn't support these templates (e.g. `Neato`) you can temporarily switch to `default`
2. Mark desired shape/point on the map
3. Press "play" button and hold it for ~1s
    ![image](https://user-images.githubusercontent.com/6118709/149677006-38e893db-da0a-4776-8a62-4a5b887668a5.png)
4. A popup containing coordinates should be shown
    ![image](https://user-images.githubusercontent.com/6118709/149677139-da76cead-c99f-4acd-be44-3191d7d1885a.png)
5. If popup doesn't appear you can enable debug mode by adding `debug: true` to card's configuration
6. If content of popup is cropped or you can't copy it, you can view coordinates in [browser's console](https://webmasters.stackexchange.com/questions/8525/how-do-i-open-the-javascript-console-in-different-browsers)
    <table>
    <tr><td>popup:</td><td>console:</td></tr>
    <tr><td>

    ![image](https://user-images.githubusercontent.com/6118709/149677314-5a41f3f2-19fd-47c5-a5cf-4d502ff3e05f.png)

    </td><td>

    ![image](https://user-images.githubusercontent.com/6118709/149677322-baaa5376-3b93-4208-ab7a-c56fa1913939.png)

    </td></tr></table>

</details>

**Comment by MrHollowX** [comment]
Awesome, thanks for this.

**Comment by bfg100k** [comment]
Dumb question - are the coordinates [x1,y1,x2,y2] or [x1,x2,y1,y2]? I suspect its the former and also x1,y1 being the top left point and x2,y2 being the bottom right point?

**Reply by PiotrMachowski** [reply]
They are `[x1,y1,x2,y2]`. `x1` and `y1` are ones with smaller values from `[x1, x2]` and `[y1, y2]` pairs respectively
